common/lib/capa/capa/responsetypes.py

```diff
@@ -720,14 +726,103 @@ def assign_choice_names(self):
 
     def get_score(self, student_answers):
 
+        """
+        Answer lists:
+          all_choices: the full list of checkboxes
+          student_answer: what the student actually chose (note no "s")
+          student_non_answer: what they didn't choose
+          self.correct_choices: boxes that should be checked
+          self.incorrect_choices: boxes that should NOT be checked
+        """
+        
+        # pipe for union of sets
+        all_choices = self.correct_choices | self.incorrect_choices   
+        
         student_answer = student_answers.get(self.answer_id, [])
 
         if not isinstance(student_answer, list):
             student_answer = [student_answer]
 
         no_empty_answer = student_answer != []
+        # "None apply" should really be a valid choice for "check all that apply",
+        # but it throws an error if all the checks are blank.
+        
         student_answer = set(student_answer)
 
+        student_non_answers = all_choices - student_answer
+        
+        if not no_empty_answer:
+            print "Empty Answer"
+            return CorrectMap(self.answer_id, 'incorrect')
+        
+        
+        """
+        This below checks to see whether we're using an alternate grading scheme.
+          Set partial_credit="false" (or remove it) to require an exact answer for any credit.
+          Set partial_credit="EDC" to count each choice for equal points (Every Decision Counts).
+          Set partial_credit="halves" to take half credit off for each error.
+        """
```

Comment:
Actually, perhaps you should move this section of code into its own method. It's very long and generally, functions should try to be 50 lines or fewer.